plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -77,14 +81,40 @@ export class TeamManager {
 
         const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
         try {
-            const team: Team | null = (await this.db.fetchTeam(teamId)) || null
+            const team: Team | null = await this.db.fetchTeam(teamId)
+            if (!team) {
+                return null
+            }
             this.teamCache.set(teamId, team)
             return team
         } finally {
             clearTimeout(timeout)
         }
     }
 
+    public async getTeamByToken(token: string): Promise<Team | null> {
+        const cachedTeamId = this.tokenToTeamIdCache.get(token)
+        if (cachedTeamId) {
+            const cachedTeam = this.teamCache.get(cachedTeamId)
+            if (cachedTeam) {
+                return cachedTeam
+            }
+        }
+
+        const timeout = timeoutGuard(`Still running "fetchTeam". Timeout warning after 30 sec!`)
+        try {
+            const team = await this.db.fetchTeamByToken(token)
+            if (!team) {
+                return null
```

Comment:
we might want to cache negative results for a short time (e.g. 1min) too, so if someone is sending us a bunch of events with bad tokens we don't hammer postgres every time.

Reply:
Yeah that's fair